From 73db44e7ac524e84e5f0fda2d60069a9e954ad1b Mon Sep 17 00:00:00 2001
From: Kristian HDiffggggggggsberg <krh@redhat.com>
Date: Wed, 11 Mar 2009 15:10:57 +0000
Subject: Drop Legacy3D option, only use fixed texture space with non-gem.

With this change, we always expect the 3D driver to use GEM textures
when the 2D driver uses GEM.  When GEM is not available or disabled,
we fall back to legacy fixed textures.
---
diff -Nurp patched/man/intel.man working/man/intel.man
--- patched/man/intel.man	2009-04-01 00:45:59.000000000 -0700
+++ working/man/intel.man	2009-04-01 00:48:15.000000000 -0700
@@ -143,13 +143,6 @@ option can be used to place the overlay 
 have options for selecting adaptors.
 Default: Textured video adaptor is preferred.
 .TP
-.BI "Option \*qLegacy3D\*q \*q" boolean \*q
-Enable support for the non-GEM mode of the 3D driver on i830 and newer.
-This will allocate a large static area for older Mesa to use for its texture
-pool.  On systems with a working GEM environment, this can be disabled to
-increase the memory pool available to other graphics tasks.
-Default for i830 and newer: Enabled.
-Default for i810: this option is not used.
 .BI "Option \*qAccelMethod\*q \*q" string \*q
 Choose acceleration architecture, either "XAA" or "EXA".  XAA is the old
 XFree86 based acceleration architecture.  EXA is a newer and simpler
diff -Nurp patched/src/i830_driver.c working/src/i830_driver.c
--- patched/src/i830_driver.c	2009-04-01 00:45:59.000000000 -0700
+++ working/src/i830_driver.c	2009-04-01 00:46:53.000000000 -0700
@@ -310,7 +310,6 @@ typedef enum {
    OPTION_LVDS24BITMODE,
    OPTION_FBC,
    OPTION_TILING,
-   OPTION_LEGACY3D,
    OPTION_LVDSFIXEDMODE,
    OPTION_TRIPLEBUFFER,
    OPTION_FORCEENABLEPIPEA,
@@ -335,9 +334,6 @@ static OptionInfoRec I830Options[] = {
    {OPTION_LVDS24BITMODE, "LVDS24Bit",	OPTV_BOOLEAN,	{0},	FALSE},
    {OPTION_FBC,		"FramebufferCompression", OPTV_BOOLEAN, {0}, TRUE},
    {OPTION_TILING,	"Tiling",	OPTV_BOOLEAN,	{0},	TRUE},
-#ifdef XF86DRI
-   {OPTION_LEGACY3D,	"Legacy3D",     OPTV_BOOLEAN,	{0},	FALSE},
-#endif
    {OPTION_LVDSFIXEDMODE, "LVDSFixedMode", OPTV_BOOLEAN,	{0},	FALSE},
    {OPTION_FORCEENABLEPIPEA, "ForceEnablePipeA", OPTV_BOOLEAN,	{0},	FALSE},
 #ifdef INTEL_XVMC
@@ -3129,11 +3125,6 @@ I830ScreenInit(int scrnIndex, ScreenPtr 
        pI830->directRenderingType = DRI_DISABLED;
    if (!pI830->can_resize && pI830->directRenderingType == DRI_NONE && I830DRIScreenInit(pScreen))
        pI830->directRenderingType = DRI_XF86DRI;
-
-   if (pI830->directRenderingType == DRI_XF86DRI) {
-       pI830->allocate_classic_textures =
-	   xf86ReturnOptValBool(pI830->Options, OPTION_LEGACY3D, TRUE);
-   }
 #endif
 
    /* Enable tiling by default */
diff -Nurp patched/src/i830_memory.c working/src/i830_memory.c
--- patched/src/i830_memory.c	2009-04-01 00:45:59.000000000 -0700
+++ working/src/i830_memory.c	2009-04-01 00:46:53.000000000 -0700
@@ -506,9 +506,6 @@ i830_allocator_init(ScrnInfoPtr pScrn, u
 	    mmsize -= ROUND_TO_PAGE(3 * pScrn->displayWidth * pI830->cpp *
 				    pScrn->virtualY);
 	}
-	/* Classic textures are fixed. */
-	if (pI830->allocate_classic_textures)
-	    mmsize -= MB(32);
 	/* Overlay and cursors, if physical, need to be allocated outside
 	 * of the kernel memory manager.
 	 */
@@ -568,6 +565,8 @@ i830_allocator_init(ScrnInfoPtr pScrn, u
 	    i830_free_memory(pScrn, pI830->memory_manager);
 	    pI830->memory_manager = NULL;
 	}
+    } else {
+	pI830->allocate_classic_textures = TRUE;
     }
 #endif /* XF86DRI */
 
